tests/baselines/reference/privateNameSneakyRuntimeException.js

```diff
@@ -0,0 +1,40 @@
+//// [privateNameSneakyRuntimeException.ts]
+// @target es6
+
+function createClass () {
+    return class {
+        #foo = 3;
+        equals(other: any) {
+            return this.#foo = other.#foo;
```

Comment:
That's fine for now, but I predict we'll get bugs asking for `any` to allow any private properties to be used -- the likelihood of some variables being `any` is just too high, especially in Javascript. How hard will that change be?

Reply:
Glad you looked into this. There is no compile-time error here. Only a runtime error if `other` isn't an `A`.

I renamed this test to `privateNameAndAny.ts` and cut some extraneous details of the example. "sneaky" was a bad name, as there is nothing sneaky about `any`.

That said, should the checker catch the following?  We can prove in the checker that it will fail at runtime, since there is no `#foo`  in an accessible private name scope:

```ts
// top level of the module
class A {
    method (other: any) { other.#foo }
}
```